cmake/Components.cmake

```diff
@@ -0,0 +1,146 @@
+include(CMakeParseArguments)
+
+set(global_prefix "${PROJECT_NAME}_")
+set(component_prefix "${PROJECT_NAME}_component_")
+
+define_property(GLOBAL PROPERTY "${global_prefix}INCLUDE_DIRS"
+  BRIEF_DOCS "Global include directories used by all components."
+  FULL_DOCS "Global include directories used by all components."
+)
+define_property(GLOBAL PROPERTY "${global_prefix}COMPONENTS"
+  BRIEF_DOCS "List all known Aikido components."
+  FULL_DOCS "List all known Aikido components."
+)
+
+#==============================================================================
+function(add_component component)
```

Comment:
That sounds like a good solution. Everything else looks good to me, so I'll make this change and then it gets my :+1: